#### Load library

In [1]:
# library
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
pd.set_option('display.max_columns', 100)

#### For these question we will use data that has been cleaned, the detailed steps are on the jupyter notebook machine learning.

In [2]:
# load data
df = pd.read_csv('df_crime.csv')

# lowering the columns name to simplify programm typing
df.columns = [x.lower() for x in df.columns]
df.head()

C:\Users\nandafa\AppData\Local\Continuum\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2785: DtypeWarning: Columns (0,2,6,12) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


incident_number  offense_code  offense_code_group offense_description  \
0      I192077559          3115  Investigate Person  INVESTIGATE PERSON   
1      I192077332           619             Larceny  LARCENY ALL OTHERS   
2      I192076660          2629          Harassment          HARASSMENT   
3      I192075386          2629          Harassment          HARASSMENT   
4      I192075335          3208       Property Lost  PROPERTY - MISSING   

  district reporting_area shooting     occurred_on_date  year  month  \
0       B3            468      NaN  2018-04-30 09:00:00  2018      4   
1      E18            496      NaN  2018-03-06 08:00:00  2018      3   
2       E5            662      NaN  2018-10-31 12:00:00  2018     10   
3       A1             96      NaN  2018-04-09 08:43:00  2018      4   
4       D4            132      NaN  2018-01-01 00:00:00  2018      1   

  day_of_week  hour    ucr_part            street        lat       long  \
0      Monday     9  Part Three       HAZLETON ST  42.279971 -71.095534   
1     Tuesday     8    Part One     HYDE PARK AVE  42.269224 -71.120853   
2   Wednesday    12    Part Two       PRIMROSE ST  42.290765 -71.130211   
3      Monday     8    Part Two      ATLANTIC AVE  42.355264 -71.050988   
4      Monday     0  Part Three  COMMONWEALTH AVE  42.353522 -71.072838   

                      location        offense_name  
0  (42.27997063, -71.09553354)  INVESTIGATE PERSON  
1  (42.26922388, -71.12085347)  LARCENY ALL OTHERS  
2  (42.29076521, -71.13021098)          HARASSMENT  
3  (42.35526402, -71.05098788)          HARASSMENT  
4  (42.35352153, -71.07283786)  PROPERTY - MISSING

#### 1. Mention which district (A/B/C/D/E, without number) has the highest crime rate!

In [3]:
df.district.value_counts()

B2          28934
C11         24345
D4          24312
B3          21477
A1          20467
C6          14391
D14         12239
E18         10875
E13         10714
E5           8639
A7           7703
A15          4111
External      320
Name: district, dtype: int64

*District B2 has a high crime rate of 28934*

#### 2. Mention 3 types of crime that happened in 2020 but didn't happen in 2019!

First, filter the data crime in 2019 and 2020, then look for the unique offense name slice from the data for the two years.

In [4]:
# filter data 2019
df_19 = df[df.year==2019]
offense_19= df_19.offense_name.unique()

In [5]:
# filter data 2020
df_20 = df[df.year==2020]
offense_20 = df_20.offense_name.unique()

In [6]:
# get the differences
list(set(offense_20) - set(offense_19))

['TRUANCY / RUNAWAY',
 'OBSCENE PHONE CALLS',
 'EXPLOSIVES - POSSESSION OR USE',
 'EXPLOSIVES - TURNED IN OR FOUND']

*Three types of crimes that occurred in 2020 but did not occur in 2019 are* 
* 'TRUANCY / RUNAWAY'
* 'EXPLOSIVES - POSSESSION OR USE'
* 'OBSCENE PHONE CALLS'* 

#### 3. What is the group of crimes that occur the most in the 40-45 week?

Change the format after_on_date to datetime format, then extract the week of year, then calculate the count per offense name

In [7]:
df['occurred_on_date'] = pd.to_datetime(df['occurred_on_date'])

In [8]:
df['week_of_year'] = df['occurred_on_date'].dt.weekofyear

C:\Users\nandafa\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:1: FutureWarning: Series.dt.weekofyear and Series.dt.week have been deprecated.  Please use Series.dt.isocalendar().week instead.
  """Entry point for launching an IPython kernel.


In [9]:
df_40_45 = df[(df['week_of_year']>=40)&(df['week_of_year']<=45)]
df_40_45.offense_name.value_counts()

INVESTIGATE PERSON                       1791
M/V - LEAVING SCENE - PROPERTY DAMAGE    1611
SICK/INJURED/MEDICAL - PERSON            1410
VANDALISM                                1263
INVESTIGATE PROPERTY                     1217
                                         ... 
CONSPIRACY EXCEPT DRUG LAW                  1
LARCENY THEFT FROM COIN-OP MACHINE          1
HOME INVASION                               1
FIREARM/WEAPON - LOST                       1
EXPLOSIVES - TURNED IN OR FOUND             1
Name: offense_name, Length: 164, dtype: int64

*The most common criminal group in the 40-45 week is INVESTIGATE PERSON*

#### 4. Mention the name of the district (A/B/C/D/E, without number) that has the lowest number of crimes on the weekend!

Filter data on Saturday only

In [10]:
df_weekend = df[df.day_of_week=='Saturday']
df_weekend.district.value_counts()

B2          3850
D4          3394
C11         3340
A1          3121
B3          3027
C6          2047
D14         1736
E18         1464
E13         1420
E5          1163
A7          1023
A15          471
External      42
Name: district, dtype: int64

*The district that has the lowest number of crimes on weekends is External*

#### 5. Mention the name of the district (A/B/C/D/E, without number) that has the highest number of criminal acts on weekdays!

Filter data is not saturday

In [11]:
df_weekday = df[df.day_of_week!='Saturday']
df_weekday.district.value_counts()

B2          25084
C11         21005
D4          20918
B3          18450
A1          17346
C6          12344
D14         10503
E18          9411
E13          9294
E5           7476
A7           6680
A15          3640
External      278
Name: district, dtype: int64

*The district that has the highest number of crimes on weekdays is B2*